OneCryptoPkg/OneCryptoBin/OneCryptoBin.c

```diff
@@ -0,0 +1,450 @@
+  CryptoProtocol->TlsSetVerifyHost           = TlsSetVerifyHost;
+  CryptoProtocol->TlsSetSessionId            = TlsSetSessionId;
+  CryptoProtocol->TlsSetCaCertificate        = TlsSetCaCertificate;
+  CryptoProtocol->TlsSetHostPublicCert       = TlsSetHostPublicCert;
+  CryptoProtocol->TlsSetHostPrivateKeyEx     = TlsSetHostPrivateKeyEx;
+  CryptoProtocol->TlsSetHostPrivateKey       = TlsSetHostPrivateKey;
+  CryptoProtocol->TlsSetCertRevocationList   = TlsSetCertRevocationList;
+  CryptoProtocol->TlsSetSignatureAlgoList    = TlsSetSignatureAlgoList;
+  CryptoProtocol->TlsSetEcCurve              = TlsSetEcCurve;
+  CryptoProtocol->TlsGetVersion              = TlsGetVersion;
+  CryptoProtocol->TlsGetConnectionEnd        = TlsGetConnectionEnd;
+  CryptoProtocol->TlsGetCurrentCipher        = TlsGetCurrentCipher;
+  CryptoProtocol->TlsGetCurrentCompressionId = TlsGetCurrentCompressionId;
+  CryptoProtocol->TlsGetVerify               = TlsGetVerify;
+  CryptoProtocol->TlsGetSessionId            = TlsGetSessionId;
+  CryptoProtocol->TlsGetClientRandom         = TlsGetClientRandom;
+  CryptoProtocol->TlsGetServerRandom         = TlsGetServerRandom;
+  CryptoProtocol->TlsGetKeyMaterial          = TlsGetKeyMaterial;
+  CryptoProtocol->TlsGetCaCertificate        = TlsGetCaCertificate;
+  CryptoProtocol->TlsGetHostPublicCert       = TlsGetHostPublicCert;
+  CryptoProtocol->TlsGetHostPrivateKey       = TlsGetHostPrivateKey;
+  CryptoProtocol->TlsGetCertRevocationList   = TlsGetCertRevocationList;
+  CryptoProtocol->TlsGetExportKey            = TlsGetExportKey;
+
+  // ========================================================================================================
+  // Timestamp Primitives
+  // ========================================================================================================
+
+  CryptoProtocol->ImageTimestampVerify = ImageTimestampVerify;
+
+  DEBUG ((DEBUG_INFO, "OneCryptoBin: CryptoInit completed - comprehensive protocol populated\n"));
+}
+
+COMMON_EXPORT_API
+EFI_STATUS
+EFIAPI
+Constructor (
+  IN ONE_CRYPTO_DEPENDENCIES  *Depends,
+  OUT VOID                    **Crypto
+  )
+{
+  //
+  // Map the provided depencencies to our global instance
+  //
+  gOneCryptoDepends = Depends;
+
+  DEBUG ((DEBUG_ERROR, "OneCryptoBin: Constructor entry called\n"));
+
+  //
+  // Allocate the required space for our ONE_CRYPTO_PROTOCOL
+  // We do not care that we might be publishing more functions than available via the library instance
+  // the library should be responsible for ensuing that the crypto functionality is acceptable before
+  // attempting to use it.
+  // 
+  *Crypto = AllocateZeroPool (sizeof (ONE_CRYPTO_PROTOCOL));
+  if (*Crypto == NULL) {
+    DEBUG ((DEBUG_ERROR, "OneCryptoBin: Failed to allocate memory for Crypto protocol\n"));
+    return EFI_OUT_OF_RESOURCES;
+  }
+
+  DEBUG ((DEBUG_ERROR, "OpenSSL Version: %a (0x%lx)\n", GetOpenSslVersionText(), (UINT64)GetOpenSslVersionNumber()));
```

Comment:
Bug: DEBUG_ERROR is printing as DEBUG_INFO. Likely due to us defining our own DEBUG values. fix this